utils/polus-bfio-util/bfio/backends.py

```diff
@@ -0,0 +1,787 @@
+        
+        for z in range(Z[0],Z[1]):
+            for y in range(Y[0]//ts,int(numpy.ceil(Y[1]/ts))):
+                y_offset = int(y * y_tile_stride)
+                ind = (x_tiles + y_offset).tolist()
+                
+                o = [self._rdr.pages[z].dataoffsets[i] for i in ind]
+                b = [self._rdr.pages[z].databytecounts[i] for i in ind]
+                
+                self.logger.debug('_chunk_indices(): offsets = {}'.format(o))
+                self.logger.debug('_chunk_indices(): bytecounts = {}'.format(b))
+                
+                offsets.extend(o)
+                bytecounts.extend(b)
+        
+        return offsets,bytecounts
+    
+    def _process_chunk(self, args):
+        
+        keyframe = self._keyframe
+        out = self._image
+        
+        w,l,d,_,_ = self._tile_indices[args[1]]
+        
+        # copy decoded segments to output array
+        segment, _, shape = keyframe.decode(*args)
+        
+        if segment is None:
+            segment = keyframe.nodata
+            
+        self.logger.debug('_process_chunk(): shape = {}'.format(shape))
+        self.logger.debug('_process_chunk(): (w,l,d) = {},{},{}'.format(w[0],l[0],d[0]))
+        
+        out[l[0]: l[0] + shape[1],
+            w[0]: w[0] + shape[2],
+            d[0],0,0] = segment.squeeze()
+        
+    def _read_image(self,X,Y,Z,C,T,output):
+        if (len(C)>1 and C[0]!=0) or (len(T)>0 and T[0]!=0):
+            raise Warning('More than channel 0 was specified for either channel or timepoint data.' + \
+                          'For the Python backend, only the first channel/timepoint will be loaded.')
+        
+        # Get keyframe and filehandle objects
+        self._keyframe = self._rdr.pages[0].keyframe
+        fh = self._rdr.pages[0].parent.filehandle
+
+        # Get binary data info
+        offsets,bytecounts = self._chunk_indices(X,Y,Z)
+        
+        self.logger.debug('read_image(): _tile_indices = {}'.format(self._tile_indices))
+        
+        if self.frontend.max_workers > 1:
+            with ThreadPoolExecutor(self.frontend.max_workers) as executor:
+                executor.map(self._process_chunk,fh.read_segments(offsets,bytecounts))
+        else:
+            for args in fh.read_segments(offsets,bytecounts):
+                self._process_chunk(args)
+
+    def close(self):
+        self._rdr.close()
+
+class PythonWriter(bfio.base_classes.AbstractWriter):
+    _page_open = False
+    _current_page = None
+    
+    logger = logging.getLogger('bfio.backends.PythonWriter')
+    
+    def __init__(self, frontend):
+        super().__init__(frontend)
+        
+        if self.frontend.C > 1:
+            self.logger.warning('The BioWriter only writes single channel ' +
+                                'images, but the metadata has {} channels. '.format(self.frontend.C) +
+                                'Setting the number of channels to 1.')
```

Comment:
You should make it clear in this warning (and the nex) what is going to happen to the channels. Are all channels beyond the first one ignored? If so, then state that.